src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { deleteTodo, patchTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { AppContext } from '../AppContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const [isEditingMode, setIsEditingMode] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const input = useRef<HTMLInputElement>(null);
+
+  const {
+    allTodos,
+    setAllTodos,
+    showError,
+    setShouldShowError,
+    loadingTodosIds,
+    setLoadingTodosIds,
+  } = useContext(AppContext);
+
+  const isLoadingState = useMemo(() => (
+    loadingTodosIds.includes(id)
+  ), [loadingTodosIds]);
+
+  const deleteTodoFromServer = useCallback(async () => {
```

Comment:
not sure about using useCallback here, since you're not passing these functions to any optimized components

Reply:
Before I fix it I want to clarify the moment :harold:

`TodoInfo` component has own states and the components rerenders every time these states are changed. If we don't `useCallback` we recreate this function every render time. Isn't it bad?